pkg/replica/replica.go

```diff
@@ -988,6 +994,50 @@ func (r *Replica) readMetadata() (bool, error) {
 	return len(r.diskData) > 0, nil
 }
 
+func (r *Replica) tryRecoverVolumeMetaFile(head string) error {
+	valid, err := r.checkValidVolumeMetaData()
+	if err != nil {
+		return err
+	}
+	if valid {
+		return nil
+	}
+
+	if head != "" {
+		r.info.Head = head
+	}
+
+	if r.info.Head == "" {
+		files, err := ioutil.ReadDir(r.dir)
+		if err != nil {
+			return err
+		}
+		for _, file := range files {
+			if strings.Contains(file.Name(), types.VolumeHeadName) {
+				r.info.Head = file.Name()
+				break
+			}
+		}
+	}
+
+	logrus.WithField("replicaInfo", r.info).Warnf("Recovering volume metafile %v", r.diskPath(volumeMetaData))
```

Comment:
Log with fields is more for general info over the scope instead.

Could you just use %v to print r.info part of the message?

Reply:
Yes, Modified. Thanks.